src/Microsoft.AspNet.Routing/RouteConstraintMatcher.cs

```diff
@@ -29,30 +30,21 @@ public static class RouteConstraintMatcher
                 var constraint = kvp.Value;
                 if (!constraint.Match(httpContext, route, kvp.Key, routeValues, routeDirection))
                 {
-                    if (routeDirection.Equals(RouteDirection.IncomingRequest)
-                        && logger.IsEnabled(LogLevel.Verbose))
+                    if (routeDirection.Equals(RouteDirection.IncomingRequest))
                     {
-                        logger.WriteValues(new RouteConstraintMatcherMatchValues()
+                        string currentRouteValues = string.Empty;
+                        if (routeValues != null)
                         {
-                            ConstraintKey = kvp.Key,
-                            Constraint = kvp.Value,
-                            Matched = false
-                        });
+                           currentRouteValues  = string.Join(", ", 
+                               routeValues.Select(rv => $"Key={rv.Key};Value={rv.Value}"));
```

Comment:
We don't want to 'encode' values for logging. Just log the value that's relevant to the constraint.